### PR TITLE
Enhance Settings

### PR DESCRIPTION
**Pull request by mightymop**
fix: link to corresponding XEP reference
feat: add more XEP Infos and collapsibles
fix: uncheck and disable checkbox for omemo plugin if libsignal is not available

close: #974

### CODE REVIEW

File: src/ui/DialogSection.ts [hunk 1]
```diff
@@ -2,9 +2,11 @@ import Navigation from './DialogNavigation'
 
 export default abstract class Section {
    private element: JQuery;
+   private button: JQuery;
+   private isCollapsible : Boolean;
 
-   constructor(protected navigation: Navigation, private title?: string) {
-
+   constructor(protected navigation: Navigation, protected collapsible?: Boolean, private title?: string) {
```
Comment (sualko) [inline src/ui/DialogSection.ts:8]: `Boolean` is a class, the correct type would be `boolean`.

File: src/ui/DialogSection.ts [hunk 2]
```diff
@@ -2,9 +2,11 @@ import Navigation from './DialogNavigation'
...
+      this.isCollapsible=collapsible;
```
Comment (sualko) [inline src/ui/DialogSection.ts:9]: This is not needed if a access modifier is added to the constructor.

File: src/ui/DialogSection.ts [hunk 3]
```diff
@@ -18,13 +20,35 @@ export default abstract class Section {
...
-         let legendElement = $('<h2>');
-         legendElement.text(this.title);
-         legendElement.appendTo(this.element);
+      if (this.isCollapsible)
+      {
+         this.button = $('<button class="jsxc-collapsible-settings-button">'+this.title+'</button>');
+         this.button.on('click',(e)=>{
+            $(e.target).toggleClass('jsxs-collapsible-settings-button-active');
+            let content = $(e.target).next();
+            if (content.css('display') === 'block') {
```
Comment (sualko) [inline src/ui/DialogSection.ts:29]: We can use css to hide/show the element. E.g. `.jsxs-collapsible-settings-button-active + div {display: block;}` or something should work.

File: src/ui/dialogs/settings.ts [hunk 4]
```diff
@@ -79,7 +79,7 @@ class ClientSection extends Section {
 
 class AccountOverviewSection extends Section {
    constructor(navigation: Navigation) {
-      super(navigation, Translation.t('Accounts'));
+      super(navigation, false, Translation.t('Accounts'));
```
Comment (sualko) [inline src/ui/dialogs/settings.ts:82]: The title is a lot more common, so adding the new parameter `collapsible` to the end would simplify a lot.

Comment (sualko) [inline src/ui/dialogs/settings.ts:82]: Also adding a default parameter of `false` would better reflect the actual behaviour.

**Review by sualko — CHANGES_REQUESTED** [review]
For me the bold buttons are breaking the design, but this is probably personal preference, but I'm wondering if we still need the inside scrollbar.

File: src/ui/DialogSection.ts (hunk 1 above)
Comment (mightymop) [reply src/ui/DialogSection.ts:8]: changed..

File: src/ui/DialogSection.ts (hunk 3 above)
Comment (mightymop) [reply src/ui/DialogSection.ts:29]: i refactored the jsxs-collapsible-settings-button class

File: src/ui/dialogs/settings.ts (hunk 4 above)
Comment (mightymop) [reply src/ui/dialogs/settings.ts:82]: done

Comment (mightymop) [reply src/ui/dialogs/settings.ts:82]: done

**Pull request merged**